### PR TITLE
TBRANDS-35: video-player-block

### PR DESCRIPTION
**Pull request by nschubach**
## Description

Updating the video-player-block to v2.00

## Jira Ticket

- [TBRANDS-35](https://arcpublishing.atlassian.net/browse/TBRANDS-35)

## Acceptance Criteria

1. I can add the Video Center Player block to a page in PageBuilder Editor.
2. I can select the video(s) to display by selecting a Schema and Content Source under the Configure Content accordion.
3. I can configure the video settings by selecting Autoplay and/or Playthrough under the Video Setting accordion. 
4. I can configure how the video display by selecting Hide Title, Hide Caption, Hide Credits under the Video Display Setting accordion.
5. I can input the Title, Description and Alert Badge content under the Display Setting accordion. 
6. I can embed a text link within the Description field.
7. I can override the default Title placement by toggling on the Display Title Beneath Video Player toggle under the Display Setting accordion. 
8. Approach:
  a. Migrate the current Video Center Player Block onto the advanced styling architecture.
  b. Create a new Commerce Theme to support the Video Center Player Block.
  c. Goal is that the Video Center Player Block can be used for both Media and Commerce customers.

** AC adjustment: https://arcpublishing.atlassian.net/browse/TBRANDS-35?focusedCommentId=741657

## Test Steps
1. Checkout this branch `git checkout tbrands-35`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/video-player-block`
3. Validate that the content matches the screens provided below

  - Published Test Video: `/2022/05/12/tree-blossoms/`
    - This is on `.env` `CONTENT_BASE=https://api.sandbox.arccommerce.arcpublishing.com`
  
  - News Test Video: `/video/2020/11/17/this-is-the-video-title/`
    - This is on `.env` `CONTENT_BASE=https://api.corecomponents.arcpublishing.com`

### Before

N/A

### After
Content Configured:
image
Inherited from global:
image
image
Auto-play and Playthrough:
image
Video Subtext options:
image
Feature Video layout:
image
Inline Video Layout:
image

## Dependencies or Side Effects

For commerce testing, use arc-commerce
https://github.com/WPMedia/arc-themes-feature-pack/pull/231

For news testing, use the-gazette:
https://github.com/WPMedia/arc-themes-feature-pack/pull/238

## Author Checklist

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm run test`, made sure all tests are passing
  - [x] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/video-player-block/features/video-player/util/formatEmbedMarkup.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/**
```
Comment (matthewroach) [inline blocks/video-player-block/features/video-player/util/formatEmbedMarkup.js]: Is this the same as the Arc Themes Components - `formatPowaVideoEmbed` function? https://github.com/WPMedia/arc-themes-components/blob/arc-themes-release-version-2.00/src/utils/format-powa-video-embed/index.js

Comment (nschubach) [reply blocks/video-player-block/features/video-player/util/formatEmbedMarkup.js]: Yep, I didn't see that... will update.

**Review by JackHowa — COMMENTED** [review]
tldr: See suggested code change for this. looks like it was a typo https://github.com/WPMedia/arc-themes-blocks/pull/1332#discussion_r870789881

----

I don't see a caption here. can you update to show the caption. pretty video though

hide and show title and credits works as expected

Screen Shot 2022-05-11 at 16 43 37

Screen Shot 2022-05-11 at 16 42 53

**Comment by JackHowa** [comment]
Looks like there's also an extra space when no credit shown. not a huge deal. I would expect "Tree Blossoms" title to show only if no credit or caption sorry small thing 

Screen Shot 2022-05-11 at 16 46 08

File: blocks/video-player-block/features/video-player/default.jsx [hunk 2]
```diff
@@ -1,182 +1,142 @@
...
-	);
-};
+	const contentSource = inheritGlobalContent ? globalContent : fetchedData;
+
+	const captionTitle = inheritGlobalContent
+		? contentSource?.headlines?.basic
+		: (description && title) || contentSource?.headlines?.basic;
+
+	const captionDescription = inheritGlobalContent
+		? contentSource?.descriptions?.basic
```
Comment (JackHowa) [inline blocks/video-player-block/features/video-player/default.jsx]:
```suggestion
		? contentSource?.description?.basic
```
Screen Shot 2022-05-11 at 17 03 14

the object that's coming in is description. that's why the content source is never showing the "These are tree blossoms" caption

File: blocks/video-player-block/features/video-player/default.jsx [hunk 3]
```diff
@@ -1,182 +1,142 @@
...
+		: (title && description) || contentSource?.descriptions?.basic;
```
Comment (JackHowa) [inline blocks/video-player-block/features/video-player/default.jsx]:
```suggestion
		: (title && description) || contentSource?.description?.basic;
```

**Review by matthewroach — CHANGES_REQUESTED** [review]
Couple of items

1. The testing instructions are relying on Production content - for testing we should be using Sandbox
2. The caption on the Feature Video - Video display style has margin to the left. Is this in the designs? Seems a little odd and out of place. I would expected it to align to the left edge
3. As this is an existing block, we are missing the News theme styles
4. We should add a story in Storybook for the new block variants - Inline/Featured

**Comment by nschubach** [comment]
> Looks like there's also an extra space when no credit shown. not a huge deal. I would expect "Tree Blossoms" title to show only if no credit or caption sorry small thing

This is actually a space added by the MediaItem component: 
image

**Comment by nschubach** [comment]
> 2. The caption on the Feature Video - Video display style has margin to the left. Is this in the designs? Seems a little odd and out of place. I would expected it to align to the left edge

Yeah, Figma design has a margin on the left of the caption in feature mode:
image

**Review by matthewroach — CHANGES_REQUESTED** [review]
@nschubach Looking very good - one small issue flagged via Chromatic, the video does not fit the container full height as it did before - https://www.chromatic.com/test?appId=5f91e4721ccaba0022a10782&id=627d977412205b004a7e3958

**Comment by nschubach** [comment]
> @nschubach Looking very good - one small issue flagged via Chromatic, the video does not fit the container full height as it did before - https://www.chromatic.com/test?appId=5f91e4721ccaba0022a10782&id=627d977412205b004a7e3958

@matthewroach This is an issue with the video component/powa player.  Powa calculates their padding to two decimal places.  And our util/formatPowaVideEmbed just uses the calculated value.  So when you request a 16/9 ratio, you get 0.5625 and powa uses a padding of 5.62% which causes a discrepancy.  I can fix the util function to floor the value (I also see there is an issue with aspect-ratio v. aspectratio this may actually be the real issue :p)
image

**Comment by nschubach** [comment]
@matthewroach : Fixed it.  I needed to change the key being passed to the embed utility. :/

**Review by matthewroach — APPROVED** [review]

**Comment by nschubach** [comment]
Oddly Chromatic is still showing a little gap, but I cannot reproduce it in any way.  I even re-ran the chromatic build.
